[docs] init list of oss projects (#10758)
ray-project/ray: doc/source/oss-users.rst

> Open source projects using Ray

Reviewer: ```suggestion
Open source libraries that run on Ray
```

==============================

> The Ray ecosystem is rapidly growing! This page lists projects and companies that are known to use Ray. This does not constitute an endorsement.

Reviewer: Let's just include open source projects. It's not useful to users to know what companies are using ray, they can see ray.io for that.

.. _ray-oss-list:

> Open source projects that run on Ray

Reviewer: ```suggestion
Open source libraries that integrate with Ray
```

====================================

> The Ray ecosystem is rapidly growing! This page lists projects that are known to use Ray. This does not constitute an endorsement.

Reviewer: ```suggestion
The Ray ecosystem is rapidly growing! This page lists libraries that integrate with Ray for distributed execution. This does not constitute an endorsement.
```

> *Using Ray?* `Fill out this form <https://forms.gle/ps4S1C955kjotxd57>`__ *and we'll be in contact to get added to the list.*

Reviewer: ```suggestion
*Using Ray?* `Fill out this form <https://forms.gle/ps4S1C955kjotxd57>`__ *and we'll be in contact to get added to the list.*
```

Author: there doesn't seem to be a change here suggested so i'll just resolve

> *Using Ray?* `Fill out this form <https://forms.gle/ps4S1C955kjotxd57>`__ *and we'll be in contact to get added to the list.*

Reviewer: ```suggestion
```

Horovod
-------

Horovod is a distributed deep learning training framework for TensorFlow, Keras, PyTorch, and Apache MXNet. The goal of Horovod is to make distributed deep learning fast and easy to use.

Github: `https://github.com/horovod/horovod/ <https://github.com/horovod/horovod/>`_

MARS
----

Mars is a tensor-based unified framework for large-scale data computation which scales Numpy, Pandas and Scikit-learn.

Github: `https://github.com/mars-project/mars/ <https://github.com/mars-project/mars/>`_

Spacy
-----
spaCy is a library for advanced Natural Language Processing in Python and Cython. It's built on the very latest research, and was designed from day one to be used in real products.

> Github: `https://github.com/explosion/spaCy/ <https://github.com/explosion/spaCy/>`_

Reviewer: Can we link to their documentation on Ray integration, here and elsewhere? The generic project link isn't that useful, it's easy enough to find by googling.

Author: I've added integration docs for those that make sense.
